tests/components/smarthab/test_config_flow.py

```diff
@@ -0,0 +1,127 @@
+        "pysmarthab.SmartHab.is_logged_in", return_value=True
+    ), patch(
+        "homeassistant.components.smarthab.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.smarthab.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_EMAIL: "mock@example.com", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "mock@example.com"
+    assert result2["data"] == {
+        CONF_EMAIL: "mock@example.com",
+        CONF_PASSWORD: "test-password",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("pysmarthab.SmartHab.async_login"), patch(
+        "pysmarthab.SmartHab.is_logged_in", return_value=False
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_EMAIL: "mock@example.com", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "wrong_login"}
+
+
+async def test_form_service_error(hass):
+    """Test we handle service errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pysmarthab.SmartHab.async_login",
+        side_effect=pysmarthab.RequestFailedException(42),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_EMAIL: "mock@example.com", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "service"}
+
+
+async def test_form_unknown_error(hass):
+    """Test we handle unknown errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pysmarthab.SmartHab.async_login", side_effect=Exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_EMAIL: "mock@example.com", CONF_PASSWORD: "test-password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown_error"}
+
+
+async def test_import(hass):
+    """Test import."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_IMPORT}
```

Comment:
We should do the import with the same data that would be passed in `async_setup`.  We should be expecting to create a config entry, and not get a form.

Reply:
`harmony/test_config_flow.py` has an example of `test_form_import`

Reply:
Thanks, should have been more careful with this one